File: src/main/java/um/haberes/rest/model/FacultadPlan.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Entity
@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "facultadId", "planId" }) })
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class FacultadPlan extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = -278690591962261599L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long facultadPlanId;

	private Integer facultadId;
	private Integer planId;
	private String nombre = "";
	
	@OneToOne
	@JoinColumn(name = "facultadId", insertable = false, updatable = false)
	private Facultad facultad;

}

File: src/main/java/um/haberes/rest/kotlin/model/FacultadPlan.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*
import um.haberes.rest.model.Auditable

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["facultadId", "planId"])])
data class FacultadPlan(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var facultadPlanId: Long? = null,

    var facultadId: Int? = null,
    var planId: Int? = null,
    var nombre: String = "",

    @OneToOne
    @JoinColumn(name = "facultadId", insertable = false, updatable = false)
    var facultad: Facultad? = null

) : Auditable()
